src/sdk/pynni/nni/compression/torch/utils/sensitivity_analysis.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import copy
+import csv
+import logging
+from collections import OrderedDict
+
+import numpy as np
+import torch.nn as nn
+
+from nni.compression.torch import LevelPruner
+from nni.compression.torch import L1FilterPruner
+from nni.compression.torch import L2FilterPruner
+
+SUPPORTED_OP_NAME = ['Conv2d', 'Conv1d']
+SUPPORTED_OP_TYPE = [getattr(nn, name) for name in SUPPORTED_OP_NAME]
+
+logger = logging.getLogger('Sensitivity_Analysis')
+logger.setLevel(logging.INFO)
+
+
+class SensitivityAnalysis:
+    def __init__(self, model, val_func, sparsities=None, prune_type='l1', early_stop=None, min_threshold=None, max_threshold=None):
+        """
+        Perform sensitivity analysis for this model.
+        Parameters
+        ----------
+        model : torch.nn.Module
+            the model to perform sensitivity analysis
+        val_func : function
+            validation function for the model. Due to
+            different models may need different dataset/criterion
+            , therefore the user need to cover this part by themselves.
+            In the val_func, the model should be tested on the validation dateset,
+            and the validation accuracy should be returned as the output of val_func.
+            There are no restrictions on the input parameters of the val_function.
+            User can use the val_args, val_kwargs parameters in analysis
+            to pass all the parameters that val_func needed.
+        sparsities : list
+            The sparsity list provided by users. This parameter is set when the user
+            only wants to test some specific sparsities. In the sparsity list, each element
+            is a sparsity value which means how much weight the pruner should prune. Take
+            [0.25, 0.5, 0.75] for an example, the SensitivityAnalysis will prune 25% 50% 75%
+            weights gradually for each layer.
+        prune_type : str
+            The pruner type used to prune the conv layers, default is 'l1',
+            and 'l2', 'fine-grained' is also supported.
+        early_stop : float
+            If this flag is set, the sensitivity analysis
+            for a conv layer will early stop when the validation metric(
+            for example, accurracy/loss) has alreay droped/raised the value
+            of early_stop (0.05 for example).
+            The default value is None, which means the analysis won't stop
+            until all given sparsities are tested.
+        min_threshold : float
+            If the validation metric returned by the val_func is lower
```

Comment:
Suggest to combine `min_threshold` and `max_threshold` into one parameter `early_stop_mode` (`min` or `max`, string type), then the logic will be simpler and more align with other parts of nni (such as tuner's optimization mode)

Reply:
Yes, it sounds reasonable, I have updated the interface, please review on the latest version, thanks~